### PR TITLE
Add script to update to the latest bits from google cloud storage.

### PR DESCRIPTION
**Pull request by ILMTitan**

### CODE REVIEW

**Comment by ILMTitan** [comment]
This will fix issue #146.

File: Google.PowerShell/ReleaseFiles/PatchFromGcs.ps1 [hunk 1]
```diff
@@ -0,0 +1,43 @@
+﻿# Copyright 2016 Google Inc. All Rights Reserved.
+# Licensed under the Apache License Version 2.0.
+#
+# Updates Cloud Tools for PowerShell module to the latest found in Google Cloud Storage bucket g-cloudsharp-unsignedbinaries
```
Comment (chrsmith) [inline Google.PowerShell/ReleaseFiles/PatchFromGcs.ps1]: End sentences with a period. Also, we should line break script files at 80 cols or so.

File: Google.PowerShell/ReleaseFiles/PatchFromGcs.ps1 [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+param($installPath) # Let a user manually select a Cloud SDK install path
```
Comment (chrsmith) [inline Google.PowerShell/ReleaseFiles/PatchFromGcs.ps1]: Perhaps put the comment on the line above for readability?

File: Google.PowerShell/ReleaseFiles/PatchFromGcs.ps1 [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    if (Test-Path $hklmPath) {
+        $installPath = Get-ItemPropertyValue $hklmPath InstallLocation
+    } elseif (Test-Path $hkcuPath) {
+        $installPath = Get-ItemPropertyValue $hkcuPath InstallLocation
+    } else {
+        Write-Error "Can not find Cloud SDK from the registry."
+        return
+    }
+}
+$installPath = $installPath -replace '"' # Registry values had quotes for some reason
```
Comment (chrsmith) [inline Google.PowerShell/ReleaseFiles/PatchFromGcs.ps1]: "had quotes for some reason" implies you don't know what is going on. (Which may be true.) But to err on the sound of sounding professional, let us just say "The Cloud SDK path is quoted." and leave it at that.

File: Google.PowerShell/ReleaseFiles/PatchFromGcs.ps1 [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+$googlePowerShellPath = Join-Path $installPath "google-cloud-sdk\platform\GoogleCloudPowerShell"
+
+if (-not (Test-Path $googlePowerShellPath)) {
+    Write-Error "Can not find Google PowerShell. $googlePowerShellPath does not exist."
```
Comment (chrsmith) [inline Google.PowerShell/ReleaseFiles/PatchFromGcs.ps1]: Consider putting '`' around `$googlePowerShellPath` for readability.

File: Google.PowerShell/ReleaseFiles/PatchFromGcs.ps1 [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    return
+}
+
+$pathToOldCmdlets = "$googlePowerShellPath\..\OldPowerShell"
```
Comment (chrsmith) [inline Google.PowerShell/ReleaseFiles/PatchFromGcs.ps1]: Can we be more descriptive than "OldPowerShell"? Perhaps "GoogleCloudPowerShell-unpatched-backup"?

File: Google.PowerShell/ReleaseFiles/PatchFromGcs.ps1 [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+if (Test-Path $pathToOldCmdlets) {
+    Remove-Item $pathToOldCmdlets -Recurse
+}
+Move-Item $googlePowerShellPath $pathToOldCmdlets
+Import-Module "$pathToOldCmdlets/GoogleCloudPowerShell.psd1"
+$bucket = Get-GcsBucket g-cloudsharp-unsignedbinaries
+$zipObject = Find-GcsObject $bucket -Prefix powershell | Sort Name -Descending | Select -First 1
```
Comment (chrsmith) [inline Google.PowerShell/ReleaseFiles/PatchFromGcs.ps1]: We definitely need a comment here to explain our selection logic, and also need to output the file path so we can at least understand _what_ was patched on the machine. We might even want to go for the gold and leave the zip file in the Cloud SDK directory, so we can inspect it later. (i.e. figure out what bits were running on the machine later.)

**Comment by chrsmith** [comment]
If it does the job it is OK, but we could definitely do a better job making the code easier to read, understand, and making it easier to diagnose things later.

**Comment by ILMTitan** [comment]
PTAL

File: Google.PowerShell/ReleaseFiles/PatchFromGcs.ps1 [hunk 7]
```diff
@@ -1,9 +1,11 @@
 ﻿# Copyright 2016 Google Inc. All Rights Reserved.
 # Licensed under the Apache License Version 2.0.
 #
-# Updates Cloud Tools for PowerShell module to the latest found in Google Cloud Storage bucket g-cloudsharp-unsignedbinaries
+# Updates Cloud Tools for PowerShell module to the latest found in 
+# Google Cloud Storage bucket g-cloudsharp-unsignedbinaries.
 
-param($installPath) # Let a user manually select a Cloud SDK install path
+# Let a user manually select a Cloud SDK install path
```
Comment (chrsmith) [inline Google.PowerShell/ReleaseFiles/PatchFromGcs.ps1]: End with a period.

File: Google.PowerShell/ReleaseFiles/PatchFromGcs.ps1 [hunk 8]
```diff
@@ -19,25 +21,30 @@ if (-not $installPath) {
...
     Remove-Item $pathToOldCmdlets -Recurse
 }
 Move-Item $googlePowerShellPath $pathToOldCmdlets
 Import-Module "$pathToOldCmdlets/GoogleCloudPowerShell.psd1"
 $bucket = Get-GcsBucket g-cloudsharp-unsignedbinaries
-$zipObject = Find-GcsObject $bucket -Prefix powershell | Sort Name -Descending | Select -First 1
-$tempFile = New-TemporaryFile
-Read-GcsObject $bucket $zipObject.Name -OutFile $tempFile -Force
 
+# Find objects in the powershell directory, and select one most recently created.
```
Comment (chrsmith) [inline Google.PowerShell/ReleaseFiles/PatchFromGcs.ps1]: We probably want to be more sophisticated. For now this is OK. But in the future we probably want a prefix like "powershell-nightly-builds/" or similar.

**Comment by chrsmith** [comment]
LGTM

**Pull request merged**